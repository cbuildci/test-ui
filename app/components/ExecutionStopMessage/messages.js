/*
 * ExecutionStopMessage Messages
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
    userStop: {
        id: 'app.components.ExecutionStopMessage.userStop',
        defaultMessage: 'Stop requested after {durationHtml} by {userHtml}',
    },
    conclusionMessage: {
        id: 'app.components.ExecutionStopMessage.conclusionMessage',
        defaultMessage: `{hasStop, select, 
            yes {Stop requested after {stopDurationHtml} by {stopUserHtml} {hasConcluded, select,
                yes {and took {conclusionDurationHtml} to stop}
                no {}
            }}
            no {Concluded after {conclusionDurationHtml}}
        }`,
    },
});
