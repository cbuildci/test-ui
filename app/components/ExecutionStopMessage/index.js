/*
 * ExecutionStopMessage
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { getDateTime } from 'utils/date';
import TimeDuration from 'components/TimeDuration';
import Time from 'components/Time';
import messages from './messages';

function ExecutionStopMessage({
    githubHost,
    stopUser,
    createTime,
    conclusionTime,
    stopRequestTime,
}) {
    const values = {
        hasConcluded: 'no',
        hasStop: 'no',
    };

    createTime = Math.round(getDateTime(createTime) / 1000);

    if (stopRequestTime != null) {
        stopRequestTime = Math.round(getDateTime(stopRequestTime) / 1000);
        values.hasStop = 'yes';

        values.stopDurationHtml = (
            <strong>
                <Time date={stopRequestTime}>
                    <TimeDuration seconds={stopRequestTime - createTime}/>
                </Time>
            </strong>
        );

        values.stopUserHtml = (
            <a href={`https://${githubHost}/${stopUser}`}>
                {stopUser}
            </a>
        );
    }

    if (conclusionTime != null) {
        conclusionTime = Math.round(getDateTime(conclusionTime) / 1000);
        values.hasConcluded = 'yes';

        values.conclusionDurationHtml = (
            <strong>
                <Time date={conclusionTime}>
                    <TimeDuration seconds={conclusionTime - (stopRequestTime || createTime)}/>
                </Time>
            </strong>
        );
    }

    return (
        <FormattedMessage
            {...messages.conclusionMessage}
            values={values}
        />
    );
}

ExecutionStopMessage.propTypes = {
    githubHost: PropTypes.string.isRequired,
    stopUser: PropTypes.string,
    createTime: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
        PropTypes.instanceOf(Date),
    ]).isRequired,
    conclusionTime: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
        PropTypes.instanceOf(Date),
    ]),
    stopRequestTime: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
        PropTypes.instanceOf(Date),
    ]),
};

ExecutionStopMessage.defaultProps = {
    stopUser: null,
    conclusionTime: null,
    stopRequestTime: null,
};

export default ExecutionStopMessage;
