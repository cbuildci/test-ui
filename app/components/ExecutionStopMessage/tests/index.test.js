import React from 'react';
import { snapshots } from '../../../../internals/testing/snapshot-util';
import ExecutionStopMessage from '../index';

describe('<ExecutionStopMessage/>', () => {
    it('should render expected JSX when has concluded', () => {
        snapshots(
            <ExecutionStopMessage
                githubHost="git.com"
                createTime={1500000000000}
                conclusionTime={1500000200000}
            />
        );
    });

    it('should render expected JSX when user requests stop', () => {
        snapshots(
            <ExecutionStopMessage
                githubHost="git.com"
                createTime={1500000000000}
                stopUser="foo"
                stopRequestTime={1500000100000}
            />
        );
    });

    it('should render expected JSX when user requests stop and has concluded', () => {
        snapshots(
            <ExecutionStopMessage
                githubHost="git.com"
                createTime={1500000000000}
                stopUser="foo"
                stopRequestTime={1500000100000}
                conclusionTime={1500000200000}
            />
        );
    });
});
