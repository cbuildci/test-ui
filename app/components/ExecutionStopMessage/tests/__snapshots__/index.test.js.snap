// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExecutionStopMessage/> should render expected JSX when has concluded 1`] = `
<FormattedMessage
  defaultMessage="{hasStop, select, 
            yes {Stop requested after {stopDurationHtml} by {stopUserHtml} {hasConcluded, select,
                yes {and took {conclusionDurationHtml} to stop}
                no {}
            }}
            no {Concluded after {conclusionDurationHtml}}
        }"
  id="app.components.ExecutionStopMessage.conclusionMessage"
  values={
    Object {
      "conclusionDurationHtml": <strong>
        <InjectIntl(Time)
          date={1500000200}
        >
          <TimeDuration
            seconds={200}
          />
        </InjectIntl(Time)>
      </strong>,
      "hasConcluded": "yes",
      "hasStop": "no",
    }
  }
/>
`;

exports[`<ExecutionStopMessage/> should render expected JSX when has concluded 2`] = `
<span>
  Concluded after 
  <strong>
    <time
      class="position-relative"
      datetime="1970-01-18T08:40:00.200Z"
      id="foobar"
      tabindex="0"
    >
      <span>
        3.3 minutes
      </span>
    </time>
  </strong>
</span>
`;

exports[`<ExecutionStopMessage/> should render expected JSX when user requests stop 1`] = `
<FormattedMessage
  defaultMessage="{hasStop, select, 
            yes {Stop requested after {stopDurationHtml} by {stopUserHtml} {hasConcluded, select,
                yes {and took {conclusionDurationHtml} to stop}
                no {}
            }}
            no {Concluded after {conclusionDurationHtml}}
        }"
  id="app.components.ExecutionStopMessage.conclusionMessage"
  values={
    Object {
      "hasConcluded": "no",
      "hasStop": "yes",
      "stopDurationHtml": <strong>
        <InjectIntl(Time)
          date={1500000100}
        >
          <TimeDuration
            seconds={100}
          />
        </InjectIntl(Time)>
      </strong>,
      "stopUserHtml": <a
        href="https://git.com/foo"
      >
        @
        foo
      </a>,
    }
  }
/>
`;

exports[`<ExecutionStopMessage/> should render expected JSX when user requests stop 2`] = `
<span>
  Stop requested after 
  <strong>
    <time
      class="position-relative"
      datetime="1970-01-18T08:40:00.100Z"
      id="foobar"
      tabindex="0"
    >
      <span>
        1.7 minutes
      </span>
    </time>
  </strong>
   by 
  <a
    href="https://git.com/foo"
  >
    @foo
  </a>
   
</span>
`;

exports[`<ExecutionStopMessage/> should render expected JSX when user requests stop and has concluded 1`] = `
<FormattedMessage
  defaultMessage="{hasStop, select, 
            yes {Stop requested after {stopDurationHtml} by {stopUserHtml} {hasConcluded, select,
                yes {and took {conclusionDurationHtml} to stop}
                no {}
            }}
            no {Concluded after {conclusionDurationHtml}}
        }"
  id="app.components.ExecutionStopMessage.conclusionMessage"
  values={
    Object {
      "conclusionDurationHtml": <strong>
        <InjectIntl(Time)
          date={1500000200}
        >
          <TimeDuration
            seconds={100}
          />
        </InjectIntl(Time)>
      </strong>,
      "hasConcluded": "yes",
      "hasStop": "yes",
      "stopDurationHtml": <strong>
        <InjectIntl(Time)
          date={1500000100}
        >
          <TimeDuration
            seconds={100}
          />
        </InjectIntl(Time)>
      </strong>,
      "stopUserHtml": <a
        href="https://git.com/foo"
      >
        @
        foo
      </a>,
    }
  }
/>
`;

exports[`<ExecutionStopMessage/> should render expected JSX when user requests stop and has concluded 2`] = `
<span>
  Stop requested after 
  <strong>
    <time
      class="position-relative"
      datetime="1970-01-18T08:40:00.100Z"
      id="foobar"
      tabindex="0"
    >
      <span>
        1.7 minutes
      </span>
    </time>
  </strong>
   by 
  <a
    href="https://git.com/foo"
  >
    @foo
  </a>
   and took 
  <strong>
    <time
      class="position-relative"
      datetime="1970-01-18T08:40:00.200Z"
      id="foobar"
      tabindex="0"
    >
      <span>
        1.7 minutes
      </span>
    </time>
  </strong>
   to stop
</span>
`;
